wdl2cwl/tests/cwl_files/blast.cwl

```diff
@@ -0,0 +1,217 @@
+    default: 100
+    type: int
+  - id: runblastn.word_size
+    default: 28
+    type: int
+  - id: runblastn.reward
+    default: 1
+    type: int
+  - id: runblastn.penalty
+    default: -2
+    type: int
+  - id: runblastn.strand
+    default: both
+    type: string
+  - id: runblastn.gapopen
+    default: 0
+    type: int
+  - id: runblastn.gapextend
+    default: 0
+    type: int
+  - id: runblastn.dust
+    default: "'20 64 1'"
+    type: string
+  - id: runblastn.max_hsps
+    type:
+      - int
+      - 'null'
+  - id: runblastn.tasks
+    default: megablast
+    type: string
+  - id: runblastn.taxids
+    type:
+      - string
+      - 'null'
+  - id: runblastn.negative_taxids
+    type:
+      - string
+      - 'null'
+  - id: runblastn.lcase_masking
+    default: false
+    type: boolean
+  - id: runblastx.max_target_seqs
+    default: 100
+    type: int
+  - id: runblastx.word_size
+    default: 6
+    type: int
+  - id: runblastx.seg
+    default: "'12 2.2 2.5'"
+    type: string
+  - id: runblastx.comp_based_stats
+    default: '2'
+    type: string
+  - id: runblastx.matrix
+    default: BLOSUM62
+    type: string
+  - id: runblastx.gapopen
+    default: 11
+    type: int
+  - id: runblastx.gapextend
+    default: 1
+    type: int
+  - id: runblastx.taxids
+    type:
+      - string
+      - 'null'
+  - id: runblastx.negative_taxids
+    type:
+      - string
+      - 'null'
+  - id: runblastx.max_hsps
+    type:
+      - int
+      - 'null'
+  - id: runblastx.lcase_masking
+    default: false
+    type: boolean
+  - id: runtblastn.max_target_seqs
+    default: 100
+    type: int
+  - id: runtblastn.word_size
+    default: 6
+    type: int
+  - id: runtblastn.seg
+    default: "'12 2.2 2.5'"
+    type: string
+  - id: runtblastn.comp_based_stats
+    default: '2'
+    type: string
+  - id: runtblastn.matrix
+    default: BLOSUM62
+    type: string
+  - id: runtblastn.gapopen
+    default: 11
+    type: int
+  - id: runtblastn.gapextend
+    default: 1
+    type: int
+  - id: runtblastn.lcase_masking
+    default: false
+    type: boolean
+  - id: runtblastn.max_hsps
+    type:
+      - int
+      - 'null'
+  - id: runtblastn.taxids
+    type:
+      - string
+      - 'null'
+  - id: runtblastn.negative_taxids
+    type:
+      - string
+      - 'null'
+  - id: runtblastx.taxids
+    type:
+      - string
+      - 'null'
+  - id: runtblastx.word_size
+    default: 3
+    type: int
+  - id: runtblastx.max_target_seqs
+    default: 100
+    type: int
+  - id: runtblastx.seg
+    default: "'12 2.2 2.5'"
+    type: string
+  - id: runtblastx.matrix
+    default: BLOSUM62
+    type: string
+  - id: runtblastx.lcase_masking
+    default: false
+    type: boolean
+  - id: runtblastx.negative_taxids
+    type:
+      - string
+      - 'null'
+  - id: runtblastx.max_hsps
+    type:
+      - int
+      - 'null'
+steps: []
+outputs:
+  - id: blast.fina_output
+    outputSource: '[inputs["runtblastx.out"] === null '
+    type: File
```

Comment:
Resulting CWL file :point_up:

Reply:
Right, IIUC, the `select_first` is returning a string for an JS expression, but the `outputSource` is only `string` or `string[]`. I think we could replace that output by an `outputBinding` and `outputEval`, but I think we don't have `outputEval` at workflow level.

The WDL Conditionals were also not translated. I think we would have to handle the conditionals and then move this `select_first` to a step of the workflow? Then use `outputSource: step_created.fina_output`?

Reply:
> The WDL Conditionals were also not translated. I think we would have to handle the conditionals and then move this select_first to a step of the workflow? Then use outputSource: step_created.fina_output?

Yes! Probably can do both in the same step